src/base/settingsstorage.cpp

```diff
@@ -204,7 +204,9 @@ QVariant SettingsStorage::loadValue(const QString &key, const QVariant &defaultV
 {
     const QString realKey = mapKey(key);
     const QReadLocker locker(&m_lock);
-    return m_data.value(realKey, defaultValue);
+    const QVariant value = m_data.value(realKey);
+    // check if retrieved value is valid
```

Comment:
Can you give me an example of how invalid QVariant can get here?

Reply:
The user edited the .ini file and somehow stores an invalid QVariant.

Reply:
Perhaps it is better that we move the check to the function where we deserialize the ini, so this would become a one-time cost.

Reply:
>The user edited the .ini file and somehow stores an invalid QVariant.

I can imagine only one case of stored invalid QVariant when there is something like `@QVariant()` written. Otherwise it is always valid since it represents valid string.
>Perhaps it is better that we move the check to the function where we deserialize the ini, so this would become a one-time cost.

But what do you expect to do if invalid QVariants are detected at this stage?

Reply:
>But what do you expect to do if invalid QVariants are detected at this stage?

Simply drop it (by skipping over it). It make no sense to have these in `QVariantHash m_data`, isn't?

I've pushed an update.

Reply:
>Simply drop it (by skipping over it). It make no sense to have these in `QVariantHash m_data`, isn't?

👍 
Hope it shouldn't be there under normal conditions.